docs: API for v113 (#6093)
langflow-ai/langflow: docs/docs/API-Reference/api-reference-api-examples.md

export LANGFLOW_API_KEY="sk-..."
```

## Base

> Use the base Langflow API for running your flow and retrieving configuration information.

Reviewer: ```suggestion
Use the base Langflow API to run your flow and retrieve configuration information.
```

   <TabItem value="curl" label="curl" default>

```curl
curl -X POST \
  "$LANGFLOW_URL/api/v1/build/$FLOW_ID/flow" \
  -H "accept: application/json" \
  -H "Content-Type: application/json" \
  -H "x-api-key: $LANGFLOW_API_KEY" \

>   -d '{"message": "hello, how are you doing?"}'

Reviewer 2: the message in the /build endpoint does not really make any difference for this endpoint, we should use input_value in this endpoint

5. The clean `end` event tells you the flow executed with no errors.
If your flow executes with errors, the `error` event handler prints the errors to the playground.

### Configure the build endpoint

> The `/build` endpoint accepts values for `start_component_id` and `stop_component_id` in the body of the command to control where the flow run will start and stop.

Reviewer 2: it's not really at the body, it's on the url a query parameter

```plain
Image file content.
```

  </TabItem>
</Tabs>

### Delete file

Delete a specific file from a flow.

This example deletes the `2024-12-30_15-42-44_image-file.png` file from Langflow.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -X 'DELETE' \
  "$LANGFLOW_URL/api/v1/files/delete/$FLOW_ID/2024-12-30_15-42-44_image-file.png" \
  -H 'accept: application/json'
```

  </TabItem>
  <TabItem value="result" label="Result">

```plain
{
  "message": "File 2024-12-30_15-42-44_image-file.png deleted successfully"
}
```

  </TabItem>
</Tabs>

## Flows

Use the `/flows` endpoint to create, read, update, and delete flows.

### Create flow

Create a new flow.

<Tabs>
   <TabItem value="curl" label="curl" default>

```curl
curl -X 'POST' \
  "$LANGFLOW_URL/api/v1/flows/" \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "name": "string2",
  "description": "string",
  "icon": "string",
  "icon_bg_color": "#FF0000",
  "gradient": "string",
  "data": {},
  "is_component": false,
  "updated_at": "2024-12-30T15:48:01.519Z",
  "webhook": false,
  "endpoint_name": "string",
  "tags": [
    "string"
  ],
  "locked": false,

>   "user_id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",

Reviewer 2: I would make it clear that the folder_id and the user_id are optional and are not required for using langflow without authentication, using without those two will make the curl work just copying from the docs without change anything

</Tabs>

## Folders

Use the `/folders` endpoint to create, read, update, and delete folders.

Folders store your flows and components.

### Read folders

Get a list of Langflow folders.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -X 'GET' \

>   '$LANGFLOW_URL/api/v1/folders/' \

Reviewer 2: those will not work on mac terminal because of simple quotes

  "description": "string",
  "components_list": [],
  "flows_list": []
}'
```

  </TabItem>
  <TabItem value="result" label="Result">

```plain
{
  "name": "new_folder_name",
  "description": "string",
  "id": "b408ddb9-6266-4431-9be8-e04a62758331",
  "parent_id": null
}
```

  </TabItem>
</Tabs>

To add flows and components at folder creation, retrieve the `components_list` and `flows_list` values from the [/api/v1/store/components](#get-all-components) and [/api/v1/flows/read](#read-flows) endpoints and add them to the request body.

Adding a flow to a folder moves the flow from its previous location. The flow is not copied.

```curl
curl -X 'POST' \
  "$LANGFLOW_URL/api/v1/folders/" \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "name": "new_folder_name",
  "description": "string",
  "components_list": [
    "3fa85f64-5717-4562-b3fc-2c963f66afa6"
  ],
  "flows_list": [
    "3fa85f64-5717-4562-b3fc-2c963f66afa6"
  ]
}'
```

### Read folder

Retrieve details of a specific folder.

To find the UUID of your folder, call the [read folders](#read-folders) endpoint.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -X 'GET' \

>   '$LANGFLOW_URL/api/v1/folders/$FOLDER_ID' \

Reviewer 2: same, simple quotes will not work

  -H 'accept: application/json'
```

  </TabItem>
  <TabItem value="result" label="Result">

```plain
[
    {
        "name": "My Projects",
        "description": "Manage your own projects. Download and upload folders.",
        "id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
        "parent_id": null
    }
]
```

  </TabItem>
</Tabs>

### Update folder

Update the information of a specific folder with a `PATCH` request.

Each PATCH request updates the folder with the values you send.
Only the fields you include in your request are updated.
If you send the same values multiple times, the update is still processed, even if the values are unchanged.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -X 'PATCH' \

>   '$LANGFLOW_URL/api/v1/folders/b408ddb9-6266-4431-9be8-e04a62758331' \

Reviewer 2: simple quotes again

  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "name": "string",
  "description": "string",
  "parent_id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "components": [
    "3fa85f64-5717-4562-b3fc-2c963f66afa6"
  ],
  "flows": [
    "3fa85f64-5717-4562-b3fc-2c963f66afa6"
  ]
}'
```

  </TabItem>
  <TabItem value="result" label="Result">

```plain
{
  "name": "string",
  "description": "string",
  "id": "b408ddb9-6266-4431-9be8-e04a62758331",
  "parent_id": null
}
```

  </TabItem>
</Tabs>

### Delete folder

Delete a specific folder.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -X 'DELETE' \

>   '$LANGFLOW_URL/api/v1/folders/$FOLDER_ID' \

Reviewer 2: simple quotes again

  </TabItem>
</Tabs>

### Download folder

Download all flows from a folder as a zip file.

The `--output` flag is optional.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -X 'GET' \
  '$LANGFLOW_URL/api/v1/folders/download/b408ddb9-6266-4431-9be8-e04a62758331' \
  -H 'accept: application/json' \
  --output langflow-folder.zip
```

  </TabItem>
    <TabItem value="result" label="Result">

```plain
The folder contents.
```

  </TabItem>
</Tabs>

### Upload folder

Upload a folder to Langflow.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -X 'POST' \

>   '$LANGFLOW_URL/api/v1/folders/upload/' \

Reviewer 2: simple quotes

    "flow_id": "01ce083d-748b-4b8d-97b6-33adbb6a528a",
    "timestamp": "2024-12-23 19:20:11 UTC",
    "sender": "Machine",
    "sender_name": "AI",
    "session_id": "01ce083d-748b-4b8d-97b6-33adbb6a528a",
    "text": "Hello! It's great to see you here! What exciting project or idea are you thinking about diving into today? Whether it's something fresh and innovative or a classic concept with a twist, I'm here to help you get started! Let's brainstorm together!",
    "files": "[]",
    "edit": false,
    "properties": {
      "text_color": "",
      "background_color": "",
      "edited": false,
      "source": {
        "id": "OpenAIModel-7AjrN",
        "display_name": "OpenAI",
        "source": "gpt-4o-mini"
      },
      "icon": "OpenAI",
      "allow_markdown": false,
      "positive_feedback": null,
      "state": "complete",
      "targets": []
    },
    "category": "message",
    "content_blocks": []
  }
]
```

  </TabItem>
</Tabs>

### Delete messages

Delete specific messages by their IDs.

This example deletes the message retrieved in the previous Get messages example.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -v -X 'DELETE' \

>   '$LANGFLOW_URL/api/v1/monitor/messages' \

Reviewer 2: simple quotes

  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '[
  "1c1d6134-9b8b-4079-931c-84dcaddf19ba"
]'
```
  </TabItem>
  <TabItem value="result" label="Result">

```plain
204 No Content
```

  </TabItem>
</Tabs>

To delete multiple messages, list the IDs within the array.

```curl
curl -v -X 'DELETE' \

>   '$LANGFLOW_URL/api/v1/monitor/messages' \

Reviewer 2: simple quotes

    "timestamp": "2024-12-23 18:49:55 UTC",
    "sender": "User",
    "sender_name": "User",
    "session_id": "different_session_id",
    "text": "message",
    "files": "[]",
    "edit": false,
    "properties": {
      "text_color": "",
      "background_color": "",
      "edited": false,
      "source": {
        "id": null,
        "display_name": null,
        "source": null
      },
      "icon": "",
      "allow_markdown": false,
      "positive_feedback": null,
      "state": "complete",
      "targets": []
    },
    "category": "message",
    "content_blocks": []
  },
]
```

  </TabItem>
</Tabs>

### Delete messages by session

Delete all messages for a specific session.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -X 'DELETE' \
  '$LANGFLOW_URL/api/v1/monitor/messages/session/different_session_id_2' \
  -H 'accept: */*'
```

  </TabItem>
  <TabItem value="result" label="Result">

```plain
HTTP/1.1 204 No Content
```

  </TabItem>
</Tabs>

### Get transactions

Retrieve all transactions (interactions between components) for a specific flow.

<Tabs>
  <TabItem value="curl" label="curl" default>

```curl
curl -X 'GET' \

>   '$LANGFLOW_URL/api/v1/monitor/transactions?flow_id=$FLOW_ID&page=1&size=50' \

Reviewer 2: simple quotes
